include/ozo/transaction.h

```diff
@@ -0,0 +1,90 @@
+#pragma once
+
+#include <ozo/connection.h>
+#include <ozo/request.h>
+
+namespace ozo {
+namespace impl {
+
+template <typename T>
+struct transaction {
+    static_assert(Connection<T>, "is not a Connection");
+
+    T conn_;
+
+    friend auto& unwrap_connection(transaction& self) noexcept {
+        return unwrap_connection(self.conn_);
+    }
+
+    friend const auto& unwrap_connection(const transaction& self) noexcept {
+        return unwrap_connection(self.conn_);
+    }
+};
```

Comment:
We need to do something at the transaction object destruction, in case the transaction is still in progress, to avoid such things as returning connection with not finished transaction into a connection pool, for example.